### PR TITLE
feat: add moehu support

### LINKED ISSUE #50: Feature: Support Moehu.org Source
### Sources

Danbooru

### Describe the problem related to the feature request

Add support for [moehu.org](https://img.moehu.org/) image source

### Describe the solution you'd like

Add support for moehu.org

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kabuda-czh**
fix #50

### CODE REVIEW

**Comment by Kabuda-czh** [comment]
python script
```python
import requests
import re
from bs4 import BeautifulSoup

"""
    2023.3.30 版本
"""

res = requests.get("https://img.moehu.org/")
bs = BeautifulSoup(res.text, 'html.parser')

menu_divs = bs.find_all(class_='menuDiv')[:-3]

# 最终返回的 json 数据
key_map = {}

for menu in menu_divs:
    menu_ul = menu.find_next('ul')
    for pre in menu_ul.find_all('pre'):
        name, id_ = re.search(r'\((.*?)→.*id=(.*?)".*?>', str(pre)).groups()
        key_map[name] = id_

```

**Review by Kabuda-czh — COMMENTED** [review]
review

File: packages/moehu/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  async get(query: ImageSource.Query): Promise<ImageSource.Result[]> {
+    // API docs: https://img.moehu.org/
+    const params = {
+      id: query.tags,
+      num: query.count,
+      return: "json"
+    }
+    const url = trimSlash(this.config.endpoint) + '?' + Object.entries(params).map(([key, value]) => `${key}=${value}`).join('&')
+    const { data } = await this.ctx.http.axios<Moehu.Response>(url)
```
Comment (Lipraty) [inline packages/moehu/src/index.ts:20]: migrate legacy `.axios()` to `.get()`

**Review by Lipraty — COMMENTED** [review]
upgrade dependencies

File: packages/moehu/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { closest } from 'fastest-levenshtein'
+import { Context, Schema, trimSlash } from 'koishi'
+import { ImageSource } from 'koishi-plugin-booru'
+import ids from './data/ids.json'
+import { Moehu } from './types'
+
+const availableTags: string[] = Object.entries(ids).map(([k, v]) => [k, v]).flat()
+
+class MoehuImageSource extends ImageSource<MoehuImageSource.Config> {
+  languages = ['en']
```
Comment (Lipraty) [inline packages/moehu/src/index.ts:10]:
```suggestion
  languages = ['en', 'zh']
```

**Review by Lipraty — CHANGES_REQUESTED** [review]

**Review by Lipraty — APPROVED** [review]

**Pull request merged**